app/femr/ui/controllers/MedicalController.java

```diff
@@ -181,22 +181,34 @@ public Result editGet(int patientId) {
 
             throw new RuntimeException();
         }
+
+        ServiceResponse<Map<String, List<String>>> tabFieldToTabMappingServiceResponse = tabService.retrieveTabFieldToTabMapping(false, false);
+        if (tabFieldToTabMappingServiceResponse.hasErrors()){
+
+            throw new RuntimeException();
+        }
+        Map<String, List<String>> tabFieldToTabMapping = tabFieldToTabMappingServiceResponse.getResponseObject();
+
+
+
         List<TabItem> tabItems = tabItemServiceResponse.getResponseObject();
         //match the fields to their respective tabs
         for (TabItem tabItem : tabItems) {
 
             switch (tabItem.getName().toLowerCase()) {
                 case "hpi":
-                    tabItem.setFields(FieldHelper.structureHPIFieldsForView(tabFieldMultiMap));
+                    tabItem.setFields(FieldHelper.structureHPIFieldsForView(tabFieldMultiMap, tabFieldToTabMapping.get("hpi")));
                     break;
                 case "pmh":
-                    tabItem.setFields(FieldHelper.structurePMHFieldsForView(tabFieldMultiMap));
+                    tabItem.setFields(FieldHelper.structurePMHFieldsForView(tabFieldMultiMap, tabFieldToTabMapping.get("pmh")));
                     break;
                 case "treatment":
-                    tabItem.setFields(FieldHelper.structureTreatmentFieldsForView(tabFieldMultiMap));
+                    tabItem.setFields(FieldHelper.structureTreatmentFieldsForView(tabFieldMultiMap, tabFieldToTabMapping.get("treatment")));
+                    break;
+                case "photos":
                     break;
                 default:
-                    tabItem.setFields(fieldHelper.structureDynamicFieldsForView(tabFieldMultiMap));
+                    tabItem.setFields(fieldHelper.structureDynamicFieldsForView(tabFieldMultiMap, tabFieldToTabMapping.get(tabItem.getName())));
                     break;
```

Comment:
The only issue I had was with line 211.

`tabItem.setFields(fieldHelper.structureDynamicFieldsForView(tabFieldMultiMap, tabFieldToTabMapping.get(tabItem.getName())));`

should be

`tabItem.setFields(fieldHelper.structureDynamicFieldsForView(tabFieldMultiMap, tabFieldToTabMapping.get(tabItem.getName().toLowerCase())));`

When the TabService is building the "TabFieldToTab" map, the response object is indexed by a lowercase name. Then on the line above, the field names are accessed by whatever was typed into the admin screen. Adding the toLowerCase fixed a null exception I got during testing.

Full disclosure: I stumbled upon this completely by accident.